runtime/src/main/cpp/CppSupport.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#ifndef RUNTIME_CPP_SUPPORT_H
+#define RUNTIME_CPP_SUPPORT_H
+
+#include <type_traits>
+#include <memory>
+
+// A collection of backported utilities from future C++ versions.
+
+namespace kotlin {
```

Comment:
Should it be `kotlin::std` namespace?

Reply:
Together with https://github.com/JetBrains/kotlin-native/pull/4525#discussion_r525952842 I think not. However I'd be fine putting it into `namespace kotlin::std_support` or something like that.